static/test/javascripts/spec/common/onward/breaking-news.spec.js

```diff
@@ -0,0 +1,286 @@
+            knownAlertIDsStorageKey = 'gu.breaking-news.hidden',
+            server;
+
+        function alertThatIs(type, options) {
+            options = defaults(options || {}, {
+                collection: 'uk',
+                age: 1
+            });
+            var sAge = options.age !== 1 ? options.age + 'min ' : '';
+            return {
+                href: options.collection,
+                content: [{
+                    headline: sAge + options.collection + ' ' + type + ' headline',
+                    trailText: options.collection + ' ' + type + ' trailText',
+                    id: options.collection + '_' + type,
+                    frontPublicationDate: Date.now() - (1000 * 60 * options.age)
+                }]
+            };
+        }
+
+        function mockBreakingNewsWith(collections) {
+            return new Promise(function (resolve, reject) {
+                injector.mock({
+                    'common/utils/storage': storage,
+                    'common/utils/ajax': ajax
+                }).require(['common/modules/onward/breaking-news'], function (breakingNews) {
+                    server = sinon.fakeServer.create();
+                    server.respondImmediately = true;
+                    server.autoRespond = true;
+                    server.respondWith(breakingNewsURL, [200, {'Content-Type': 'application/json'}, JSON.stringify({
+                        webTitle: 'Breaking News',
+                        collections: collections
+                    })]);
+                    breakingNews().then(function(result) {
+                        // make sure the DOM has finished updating
+                        requestAnimationFrame(function () {
+                            resolve(result);
+                        });
+                    }, reject);
+                    server.restore();
+                }, function (e) {
+                    console.log(e);
+                });
+            });
+        }
+
+        beforeAll(function () {
+            ajax = jasmine.createSpy().and.callFake(ajax);
+            config.page.edition = 'UK';
+        });
+
+        beforeEach(function (done) {
+            $('body').html('<div class="js-breaking-news-placeholder breaking-news breaking-news--hidden breaking-news--fade-in" data-link-name="breaking news" data-component="breaking-news"></div>');
+            requestAnimationFrame(done);
+        });
+
+        describe('user cannot dismiss alerts', function () {
+            beforeEach(function () {
+                sinon.stub(storage.local, 'isAvailable').returns(false);
+            });
+
+            it('should not try and fetch the json', function (done) {
+                mockBreakingNewsWith([]).then(function () {
+                    fail('user cannot use local storage, but we seem to think things are okish');
+                    done();
```

Comment:
Not worth changing it all now, but I wonder whether we could use `.finally(done)` (if we're using when.js) as you have it in all `resolve` and `reject`. I am not sure why the es6 implementation doesn't have that method though. Apparently you could do `.then(done)` (after the catch if there is one): https://gist.github.com/jish/e9bcd75e391a2b21206b (And seems like Matt Andrews had this though at some point too... https://github.com/matthew-andrews/Promise.prototype.finally)

Anyway, you have lots of these so I am more talking out loud than recommending changing it :)